src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -450,6 +450,192 @@ private Exception FlattenIfAggregatedException(Exception ex)
             return ex;
         }
 
+        protected async Task<IActionResult> GetDiagnosticReport(TResource resource, DiagnosticReportQuery queryBody, DateTime startTime, DateTime endTime, TimeSpan timeGrain, string correlationId=null)
+        {
+            if (correlationId==null)
+            {
+                correlationId = Guid.NewGuid().ToString();
+            }
+            DiagnosticReportEnvelope response = new DiagnosticReportEnvelope()
+            {
+                CorrelationId = correlationId
+            };
+
+            RuntimeContext<TResource> cxt = PrepareContext(resource, startTime, endTime);
+            var allDetectors = await ListDetectorsInternal(cxt);
+            // Get detectors to run based on the parameters specified in the request
+            var detectorsToRun = await GetDetectorsToRun(cxt, queryBody, allDetectors);
+
+            // If any of the detectors in the list are analysis, get its children as well
+            var allChildrenOfAnalyses = new List<DiagnosticApiResponse>();
+            var detectorInsights = new List<DiagnosticReportInsight>();
+            foreach (var detector in detectorsToRun)
+            {
+                if (detector.Metadata.Type == DetectorType.Analysis)
+                {
+                    var children = InsightsAPIHelpers.GetChildrenOfAnalysis(detector.Metadata.Id, allDetectors);
+                    foreach (var child in children)
+                    {
+                        if (!detectorsToRun.Contains(child) && !allChildrenOfAnalyses.Contains(child))
+                        {
+                            allChildrenOfAnalyses.Add(child);
+                        }
+                    }
+                }
+            }
+
+            var filteredDetectorsToRun = detectorsToRun.Union(allChildrenOfAnalyses).ToList();
+            DiagnosticsETWProvider.Instance.LogRuntimeHostInsightCorrelation(cxt.OperationContext.RequestId, "DiagnosticReportAPI.DetectorsRun", cxt.OperationContext.Resource.SubscriptionId,
+                    cxt.OperationContext.Resource.ResourceGroup, cxt.OperationContext.Resource.Name, correlationId, $"Running detectors: {JsonConvert.SerializeObject(filteredDetectorsToRun.Select(x => x.Metadata.Id).ToList())}");
+
+            if (filteredDetectorsToRun.Count>0)
```

Comment:
nitpick - just format the whole document once :)

Reply:
Done